compiler/rustc_type_ir/src/interner.rs

```diff
@@ -262,6 +262,8 @@ pub trait Interner:
 
     fn trait_is_alias(self, trait_def_id: Self::DefId) -> bool;
 
+    fn trait_is_async_fn(self, trait_def_id: Self::DefId) -> bool;
```

Comment:
I don't think we need this helper. Can you replace this call with a call to `async_fn_trait_kind_from_def_id(def_id).is_some()`

Reply:
Unless you want to clean up *all* of the usages of `async_fn_trait_kind_from_def_id` and `fn_trait_kind_from_def_id` , but I would kinda prefer if that was done in a separate PR or at least a separate commit.

But even if we did that, I'd probably bikeshed the name a bit. `is_async_fn` makes it seem like it's == AsyncFn, not "async fn trait family".